packages/app/src/cli/models/extensions/schemas.ts

```diff
@@ -18,6 +18,11 @@ export const TypeSchema = zod.object({
   type: zod.string().default('ui_extension'),
 })
 
+export const ExtensionsArraySchema = zod.object({
+  type: zod.string().optional(),
```

Comment:
I have the same question as @MaxLeduc about `type` and `handle`, should they be on each extension in the array, or here at the higher level? https://github.com/Shopify/cli/pull/2366#discussion_r1258910061

Reply:
this specific `type` is to support extensions with the legacy format. Extensions using the unified config will have `type` and `handle` in each extension

Reply:
I see - so right now the flow specs still have handle at the top level (i.e. `config.handle`), and later on it could be inside each extension?

Reply:
With this change, all extensions will receive a flattened config, so flow doesn't know anymore about the "extensions array", so from the specification point of view, everything is "top level". So yes, from the spec the correct way of accessing the handle is `config.handle`, but in the toml it will be part of the extensions array object (not at the top)